src/main/java/SaveFile.java

```diff
@@ -0,0 +1,93 @@
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.TaskList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.stream.StreamSupport;
+
+public class SaveFile {
+
+    protected String path;
+
+    public SaveFile(String path){
+        this.path = path;
+    }
+
+    public TaskList read() throws IOException{
+
+        File t = new File(this.path);
+
+        TaskList taskArrayList = new TaskList();
+        if(t.exists()){
+            try{
+                Scanner scanner = new Scanner(t);
+                int index = 0;
+                while(scanner.hasNext()){
+                    String[] parts = scanner.nextLine().split("\\|");
+                    String type = parts[0];
+                    String status = parts[1];
+                    String description = parts[2];
+                    switch (type) {
+                        case "T":
+                            taskArrayList.add(new Task(description));
```

Comment:
There are indentation for each of your `case` clauses. This deviates from the [Java coding standard](https://se-education.org/guides/conventions/java/basic.html#layout) convention for switch case.